pkg/controller/xgboostjob/xgboostjob_controller.go

```diff
@@ -112,55 +156,83 @@ func (r *ReconcileXGBoostJob) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
-	// TODO(user): Change this to be the object type created by your controller
-	// Define the desired Deployment object
-	deploy := &appsv1.Deployment{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      instance.Name + "-deployment",
-			Namespace: instance.Namespace,
-		},
-		Spec: appsv1.DeploymentSpec{
-			Selector: &metav1.LabelSelector{
-				MatchLabels: map[string]string{"deployment": instance.Name + "-deployment"},
-			},
-			Template: corev1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"deployment": instance.Name + "-deployment"}},
-				Spec: corev1.PodSpec{
-					Containers: []corev1.Container{
-						{
-							Name:  "nginx",
-							Image: "nginx",
-						},
-					},
-				},
-			},
-		},
+	// Check reconcile is required.
+	needSync := r.satisfiedExpectations(xgbjob)
+
+	if !needSync || xgbjob.DeletionTimestamp != nil {
+		log.Info("reconcile cancelled, job does not need to do reconcile or has been deleted",
+			"sync", needSync, "deleted", xgbjob.DeletionTimestamp != nil)
+		return reconcile.Result{}, nil
 	}
-	if err := controllerutil.SetControllerReference(instance, deploy, r.scheme); err != nil {
+	oldStatus := xgbjob.Status.DeepCopy()
+	// Set default priorities for xgboost job
+	scheme.Scheme.Default(xgbjob)
+
+	// Use common to reconcile the job related pod and service
+	err = r.xgbJobController.ReconcileJobs(xgbjob, xgbjob.Spec.XGBReplicaSpecs, xgbjob.Status.JobStatus, &xgbjob.Spec.RunPolicy)
+
+	if err != nil {
 		return reconcile.Result{}, err
 	}
+	if !reflect.DeepEqual(oldStatus, &xgbjob.Status.JobStatus) {
+		err = r.UpdateJobStatusInApiServer(xgbjob, &xgbjob.Status.JobStatus)
+	}
+	return reconcile.Result{}, err
+}
 
-	// TODO(user): Change this for the object type created by your controller
-	// Check if the Deployment already exists
-	found := &appsv1.Deployment{}
-	err = r.Get(context.TODO(), types.NamespacedName{Name: deploy.Name, Namespace: deploy.Namespace}, found)
-	if err != nil && errors.IsNotFound(err) {
-		log.Info("Creating Deployment", "namespace", deploy.Namespace, "name", deploy.Name)
-		err = r.Create(context.TODO(), deploy)
-		return reconcile.Result{}, err
-	} else if err != nil {
-		return reconcile.Result{}, err
+// UpdateJobStatusInApiServer updates the job status in to cluster.
+func (r *ReconcileXGBoostJob) UpdateJobStatusInApiServer(job interface{}, jobStatus *v1.JobStatus) error {
```

Comment:
s/UpdateJobStatusInApiServer/UpdateJobStatusInAPIServer

Reply:
this is the interface of common